### PR TITLE
chore: refactor .layout-header into separate component

### PR DESCRIPTION
**Pull request by katiegoines**
#### Description of changes:
Refactor the elements in .layout-header to separate LayoutHeader own component

staging: https://layout-component-refactor.d1egzztxsxq9xz.amplifyapp.com/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/components/Layout/LayoutHeader.tsx
Comment (hbuchel) [inline src/components/Layout/LayoutHeader.tsx:1]: Just a reminder to clean up some of the commented out bits in this file (the import, handleScroll, and useEffect).

**Review by hbuchel — COMMENTED** [review]
Just a note about some commented out code. And assuming we can merge your Breadcrumbs PR first which will remove these Breadcrumbs file changes (though I think they're the same?). Otherwise LGTM! Probably we can piece this out a bit more but this definitely makes the file more work-able.

File: src/components/Layout/LayoutHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  platform,
+  showLastUpdatedDate = true
+}: {
+  hasTOC?: boolean;
+  tocHeadings: HeadingInterface[];
+  pageType?: 'home' | 'inner';
+  platform?: Platform;
+  showLastUpdatedDate: boolean;
+}) => {
+  const [menuOpen, toggleMenuOpen] = useState(false);
```
Comment (hbuchel) [inline src/components/Layout/LayoutHeader.tsx:36]: Can you take a look at this functionality? The opening/closing of the menu works on mobile, but closing the menu when you navigate to a page doesn't occur now (currently on prod if you are in the mobile view, you click on a page in the menu, the menu closes). `toggleMenuOpen` is part of the context passed down from LayoutProvider, maybe we need to import that context here like we do in [MenuItem](https://github.com/aws-amplify/docs/blob/main/src/components/Menu/MenuItem.tsx#L43)

**Review by hbuchel — APPROVED** [review]
Thanks @katiegoines!

**Review by timngyn — APPROVED** [review]

**Pull request merged**